pkg/endpoint/public_ip.go

```diff
@@ -104,11 +105,15 @@ func getPublicIP(submSpec *types.SubmarinerSpecification, k8sClient kubernetes.I
 		}
 
 		// If this resolver failed, we log it, but we fall back to the next one
-		logger.V(log.DEBUG).Infof("Error resolving public IP with resolver %s, config: %s: %v", resolver, config, err)
+		if wrappedErr == nil {
+			wrappedErr = errors.Wrapf(err, "\nResolver[%q]", resolver)
+		} else {
+			wrappedErr = errors.Wrapf(wrappedErr, "\nResolver[%q]: %v", resolver, err.Error())
+		}
```

Comment:
Rather than wrapping each error in a nested manner, perhaps use [NewAggregate](https://github.com/kubernetes/apimachinery/blob/8d1258da8f386b809d312cdda316366d5612f54e/pkg/util/errors/errors.go#L46) instead - a composite/aggregate error seems more suitable here..